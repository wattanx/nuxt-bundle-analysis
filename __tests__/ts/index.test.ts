import { execSync } from "child_process";
import fs from "fs";

describe("ts test", () => {
  it("generated __bundle_analysis.json", async () => {
    execSync("yarn report:ts");
    expect(
      fs.existsSync("../sample/.nuxt/analyze/__bundle_analysis.json")
    ).toBeTruthy();
  });

  it("generated __bundle_analysis_comment.txt", () => {
    execSync("yarn compare:ts");
    expect(
      fs.existsSync("../sample/.nuxt/analyze/__bundle_analysis_comment.txt")
    ).toBeTruthy();
  });

  it("comparison results test", () => {
    const contents = fs
      .readFileSync("../sample/.nuxt/analyze/__bundle_analysis_comment.txt")
      .toString();

    const splitContents = contents.split("\n");

    // decreased
    expect(splitContents.find((x) => x.includes("commons/app"))).toContain(
      "ðŸŸ¢"
    );

    // increased
    expect(
      splitContents.find((x) => x.includes("components/tutorial"))
    ).toContain("ðŸ”´");

    // removed
    expect(splitContents.find((x) => x.includes("pages/test"))).toContain(
      "removed"
    );

    // added
    expect(
      splitContents.find((x) => x.includes("components/nuxt-logo"))
    ).toContain("added");

    expect(contents).toMatchSnapshot();
  });

  it("Snapshot of comparison results", () => {
    const contents = fs
      .readFileSync("../sample/.nuxt/analyze/__bundle_analysis_comment.txt")
      .toString();

    expect(contents).toMatchSnapshot();
  });
});
