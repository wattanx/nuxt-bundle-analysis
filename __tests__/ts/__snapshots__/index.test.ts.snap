// Vitest Snapshot v1

exports[`js test > Snapshot of comparison results 1`] = `
"
<!-- __NUXTJS_BUNDLE -->
# Bundle Size
| Route | Size (gzipped) |
| --- | --- |
| \`app\` | 86.06 kB (游릭 -3.94 kB) |
| \`pages/index\` | 3.01 kB (游댮 1 kB) |
| \`components/nuxt-logo\` | 1.01 kB (added) |
| \`pages/test\` | removed |"
`;

exports[`js test > comparison results test 1`] = `
"
<!-- __NUXTJS_BUNDLE -->
# Bundle Size
| Route | Size (gzipped) |
| --- | --- |
| \`app\` | 86.06 kB (游릭 -3.94 kB) |
| \`pages/index\` | 3.01 kB (游댮 1 kB) |
| \`components/nuxt-logo\` | 1.01 kB (added) |
| \`pages/test\` | removed |"
`;

exports[`ts test > Snapshot of comparison results 1`] = `
"
<!-- __NUXTJS_BUNDLE -->
# Bundle Size
| Route | Size (gzipped) |
| --- | --- |
| \`app\` | 86.06 kB (游릭 -3.94 kB) |
| \`pages/index\` | 3.01 kB (游댮 1 kB) |
| \`components/nuxt-logo\` | 1.01 kB (added) |
| \`pages/test\` | removed |"
`;

exports[`ts test > comparison results test 1`] = `
"
<!-- __NUXTJS_BUNDLE -->
# Bundle Size
| Route | Size (gzipped) |
| --- | --- |
| \`app\` | 86.06 kB (游릭 -3.94 kB) |
| \`pages/index\` | 3.01 kB (游댮 1 kB) |
| \`components/nuxt-logo\` | 1.01 kB (added) |
| \`pages/test\` | removed |"
`;
